aloha-io-avro/src/test/scala/com/eharmony/aloha/semantics/compiled/plugin/avro/CompiledSemanticsAvroPluginTest.scala

```diff
@@ -0,0 +1,251 @@
+
+import scala.collection.JavaConversions.{asJavaCollection, asScalaBuffer, asScalaIterator}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+/**
+  * Created by ryan.
+  */
+@RunWith(classOf[BlockJUnit4ClassRunner])
+class CompiledSemanticsAvroPluginTest {
+  import CompiledSemanticsAvroPluginTest._
+
+  @Test def testCompilation(): Unit = {
+    val plugin = CompiledSemanticsAvroPlugin[GenericRecord](UserSchema, dereferenceAsOptional = false)
+    val semantics = CompiledSemantics(
+      compiler = new TwitterEvalCompiler(),
+      plugin = plugin,
+      imports = Seq.empty)
+
+    val user = users(Seq(
+      User(
+        name = Option("ryan"),
+        kidsAges = Seq(7, 3),
+        profile = Option(Profile(
+          height = Some(72),
+          photos = Photos(Some(6)))))
+    )).head
+
+    val name = semantics.createFunction[Option[CharSequence]]("Option(${name})", Some(None)).right.get
+    val kidAges = semantics.createFunction[Seq[Int]]("${kid_ages}").right.get
+    val profile = semantics.createFunction[Option[GenericRecord]]("Option(${profile})", Some(None)).right.get
+    val height = semantics.createFunction[Option[Int]]("Option(${profile.height})", Some(None)).right.get
+    val photoId = semantics.createFunction[Option[Long]]("Option(${profile.photos.id})", Some(None)).right.get
+    val firstKidAge = semantics.createFunction[Int]("${kid_ages[0]}").right.get
+
+    assertEquals("ryan", name(user).get.toString)
+    assertEquals(Seq(7, 3), kidAges(user))
+    assertTrue(profile(user).isDefined)
+    assertEquals(72, height(user).get)
+    assertEquals(6L, photoId(user).get)
+    assertEquals(7, firstKidAge(user))
+  }
+
+  @Test def test1(): Unit = {
+    val plugin = CompiledSemanticsAvroPlugin[GenericRecord](UserSchema)
+
+    val userRecords = users(Seq(
+      User(
+        name = Option("ryan"),
+        kidsAges = Seq(3),
+        profile = Option(Profile(height = None, photos = Photos(None)))
+      )
+    ))
+
+    val kidAges = plugin.accessorFunctionCode("kid_ages")
+    val name = plugin.accessorFunctionCode("name")
+    val profile = plugin.accessorFunctionCode("profile")
+    val height = plugin.accessorFunctionCode("profile.height")
+    val photos = plugin.accessorFunctionCode("profile.photos")
+    val photoId = plugin.accessorFunctionCode("profile.photos.id")
+    val a = 1
+
+//    Seq(kidAges, name, profile, height, photos, photoId) foreach { c =>
+//      println(c.right.get.pretty + "\n\n")
+//    }
+
+//    println(s"${name.right.get.pretty}\n$profile\n$height\n$photos\n$photoId")
+
+    val user: GenericRecord = new GenericData.Record(UserSchema)
+
+    // Apparently, the default is present but not automatically populated in a GenericRecord
+    Console.println(user.getSchema.getField("name").defaultVal())
+
+    user.getSchema.getField("kid_ages")
+
+    Console.println("kid ages: " + user.get("kid_ages"))
+
+    Console.println(user.get("name"))
+
+//    val kidAgesFn =
```

Comment:
Will remove these tests.  They're not useful anyway.  Only the AvroLikeProto one, which I'll rename.